runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -199,11 +199,46 @@ void fillTensorInfo(nnfw_tensorinfo *ti, const onert::ir::Shape &shape,
 
 nnfw_session::nnfw_session()
   : _model{nullptr}, _coptions{nullptr}, _compiler_artifact{nullptr}, _execution{nullptr},
-    _kernel_registry{std::make_shared<onert::api::CustomKernelRegistry>()}
+    _kernel_registry{nullptr}
 {
   // DO NOTHING
 }
 
+NNFW_STATUS nnfw_session::create(nnfw_session **session)
+{
+  if (session == nullptr)
+    return NNFW_STATUS_UNEXPECTED_NULL;
+
+  // Create session
+  try
+  {
+    *session = new nnfw_session();
```

Comment:
What about using smart pointer and set `*session` at the end of this function. There may be memory leak if exception is thrown between line 228-229.

Reply:
We need free on catch blocks. But it's hard to use `unique_ptr` because we can't find proper owner of new session's `unique_ptr`.

Reply:
I am not saying returning `unique_ptr`, but suggesting use `unique_ptr` as local variable in this funciton, and return raw pointer from this `unique_ptr`. Anyway, I am okay.

Reply:
>return raw pointer from this `unique_ptr`

My comment means that after returning raw pointer, allocated session by local `unique_ptr` will be deleted, so returned raw pointer will be invalidate.

Reply:
@hseok-oh Please find my comment below (I wrote yesterday for approval). I was thinking of `release()`, which doesn't free the pointer.

Reply:
I understood. Thanks.